### DISCUSSION TITLE
Is there a way to set the behavior of GDB's next instruction to be similar to LLDB's' next '?

### DISCUSSION THREAD

**Discussion by WittonBell** [microsoft/vscode-cpptools#12550] [Q&A]
When we use the GDB debugger in VS/VSCode, 'step over' skips other function calls on the same line as the previous call to this function，for example:
```cpp
int f()
{
	return 1;
} // press `F10` in VS/VSCode run `step over` instruction at this line, will skip `foo`

void foo(int i)
{
	printf("%d\n", i);
}

int main()
{
    foo(f());
    return 0;
}
```

When use lldb debugger or VC debugger, it can return at line foo(f()), and can be executed `step in` to enter `foo` function. This behavior is very useful for debugging multiple function calls on the same line. for example:
```cpp
a().b().c();
std::shared_ptr<XXX> ptr = std::make_shared<XXX>();
ptr->f().foo();
```
Is there a way to change the behavior of GDB to be similar to LLDB through a certain configuration?

**Comment by gcampbell-msft** [comment]
I believe this is more practical on the cpptools discussion, since we don't control the debugger with this extension. @sean-mcmanus Do you have any guidance or insight here?
